### PR TITLE
watch create repo

### PR DESCRIPTION
**Pull request by yangwei999**

### CODE REVIEW

**Comment by qinsheng99** [comment]
lgtm

File: softwarepkg/domain/event.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package domain
 
 import "encoding/json"
 
+const platform = "gitee"
```
Comment (zengchen1024) [inline softwarepkg/domain/event.go:5]: s/platform/platformGitee

File: softwarepkg/domain/repository/pr.go [hunk 2]
```diff
@@ -4,5 +4,8 @@ import "github.com/opensourceways/robot-gitee-software-package/softwarepkg/domai
 
 type PullRequest interface {
 	Add(*domain.PullRequest) error
+	Save(*domain.PullRequest) error
 	Find(int) (domain.PullRequest, error)
+	FindAll() ([]domain.PullRequest, error)
```
Comment (zengchen1024) [inline softwarepkg/domain/repository/pr.go:9]: no filter for FindAll ?

File: softwarepkg/domain/repository/pr.go [hunk 3]
```diff
@@ -4,5 +4,8 @@ import "github.com/opensourceways/robot-gitee-software-package/softwarepkg/domai
...
+	Remove(*domain.PullRequest) error
```
Comment (zengchen1024) [inline softwarepkg/domain/repository/pr.go:10]: Remove(num int) error

File: softwarepkg/domain/watching/repo.go [hunk 4]
```diff
@@ -1,7 +1,5 @@
 package watching
 
-import "github.com/opensourceways/robot-gitee-software-package/softwarepkg/domain"
-
 type Watching interface {
-	Apply(*domain.SoftwarePkgRepo) error
+	Run()
```
Comment (zengchen1024) [inline softwarepkg/domain/watching/repo.go:4]: no need for Watching

File: softwarepkg/infrastructure/watchingimpl/config.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package watchingimpl
+
+type Config struct {
+	Org      string `json:"org"`
+	Interval int    `json:"interval"`
```
Comment (zengchen1024) [inline softwarepkg/infrastructure/watchingimpl/config.go:5]: what's the unit for interval? comment it.

File: softwarepkg/infrastructure/watchingimpl/repo.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+type WatchingImpl struct {
+	cfg       Config
+	cli       iClient
+	log       *logrus.Entry
+	repo      repository.PullRequest
+	prService app.PullRequestService
+}
+
+type iClient interface {
```
Comment (zengchen1024) [inline softwarepkg/infrastructure/watchingimpl/repo.go:28]: move to line 19

File: softwarepkg/infrastructure/watchingimpl/config.go (hunk 5 above)
Comment (zengchen1024) [reply softwarepkg/infrastructure/watchingimpl/config.go:5]: add a func to convert interval to time.Duration

File: softwarepkg/infrastructure/watchingimpl/repo.go [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	return &WatchingImpl{
+		cfg: cfg,
+		cli: cli,
+	}
+}
+
+type WatchingImpl struct {
+	cfg       Config
+	cli       iClient
+	log       *logrus.Entry
```
Comment (zengchen1024) [inline softwarepkg/infrastructure/watchingimpl/repo.go:23]: use logrus directly

File: message-server/config.go [hunk 8]
```diff
@@ -11,4 +11,5 @@ type Topics struct {
 	ApprovedPkg  string `json:"approved_pkg" required:"true"`
 	RejectedPkg  string `json:"rejected_pkg" required:"true"`
 	AbandonedPkg string `json:"abandoned_pkg" required:"true"`
+	RepoCreated  string `json:"repo_created" required:"true"`
```
Comment (zengchen1024) [inline message-server/config.go:14]: CreatedRepo

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

File: softwarepkg/infrastructure/watchingimpl/config.go [hunk 9]
```diff
@@ -14,3 +17,7 @@ func (cfg *Config) SetDefault() {
 		cfg.Interval = 10
 	}
 }
+
+func (cfg *Config) GetInterval() time.Duration {
```
Comment (zengchen1024) [inline softwarepkg/infrastructure/watchingimpl/config.go:21]: s/GetIntervalj/IntervalDuration

File: softwarepkg/infrastructure/watchingimpl/repo.go [hunk 10]
```diff
@@ -17,23 +17,22 @@ func NewWatchingImpl(cfg Config, cli iClient) *WatchingImpl {
...
-	GetRepo(org, repo string) (sdk.Project, error)
-}
-
 func (impl *WatchingImpl) Run() {
 	for {
-		prs, err := impl.repo.FindAll()
+		prs, err := impl.repo.FindAll(true)
 		if err != nil {
-			impl.log.WithError(err).Error("find all storage pr failed")
+			logrus.WithError(err).Error("find all storage pr failed")
```
Comment (zengchen1024) [inline softwarepkg/infrastructure/watchingimpl/repo.go:35]: logrus.Errorf("find all storage pr failed, err:%s", err.Error())

**Review by zengchen1024 — CHANGES_REQUESTED** [review]

**Pull request merged**